scripts/o3de/o3de/utils.py

```diff
@@ -554,3 +706,39 @@ def replace_dict_keys_with_value_key(input:dict, value_key:str, replaced_key_nam
             input[value[value_key]] = entries
         else:
             input[value[value_key]] = value
+
+def safe_kill_processes(*processes: List[Popen], process_logger: logging.Logger = None) -> None:
+    """
+    Kills a given process without raising an error
+    :param processes: An iterable of processes to kill
+    :param process_logger: (Optional) logger to use
+    """
+    def on_terminate(proc) -> None:
+        try:
+            process_logger.info(f"process '{proc.args[0]}' with PID({proc.pid}) terminated with exit code {proc.returncode}")
+        except psutil.AccessDenied:
+            process_logger.warning("Termination failed, Access Denied with stacktrace:", exc_info=True)
+        except psutil.NoSuchProcess:
+            process_logger.warning("Termination request ignored, process was already terminated during iteration with stacktrace:", exc_info=True)
+
+    if not process_logger:
+        process_logger = logger
+    
+    for proc in processes:
+        try:
+            process_logger.info(f"Terminating process '{proc.args[0]}' with PID({proc.pid})")
+            proc.kill()
+        except psutil.AccessDenied:
+            process_logger.warning("Termination failed, Access Denied with stacktrace:", exc_info=True)
+        except psutil.NoSuchProcess:
+            process_logger.warning("Termination request ignored, process was already terminated during iteration with stacktrace:", exc_info=True)
+        except Exception:  # purposefully broad
+            process_logger.error("Unexpected exception ignored while terminating process, with stacktrace:", exc_info=True)
+    try:
+        psutil.wait_procs(processes, timeout=30, callback=on_terminate)
+    except psutil.AccessDenied:
+        process_logger.warning("Termination failed, Access Denied with stacktrace:", exc_info=True)
+    except psutil.NoSuchProcess:
+        process_logger.warning("Termination request ignored, process was already terminated during iteration with stacktrace:", exc_info=True)
+    except Exception:  # purposefully broad
+        process_logger.error("Unexpected exception while waiting for processes to terminate, with stacktrace:", exc_info=True)
```

Comment:
NIT: Looks like all of the the new code files aren't showing new lines at EoF in this PR.